File: 1736.py
class Solution:
    def maximumTime(self, time: str) -> str:
        letters = list(time)
        if letters[0] == '?':
            letters[0] = '2' if letters[1] == '?' or int(letters[1]) < 4 else '1'
        if letters[1] == '?':
            letters[1] = '3' if letters[0] == '2' else '9'
        if letters[3] == '?':
            letters[3] = '5'
        if letters[4] == '?':
            letters[4] = '9'
        return ''.join(letters)

File: sol/1736/1736.cpp
struct Solution {
    string& maximumTime(string& time) {
        if (time[0] == '?'){
            if (time[1] == '?' || ('0' <= time[1] && time[1] <= '3')){
                time[0] = '2';
            } else {
                time[0] = '1';
            }
        }
        if (time[1] == '?'){
            time[1] = time[0] == '2' ? '3' : '9';
        }
        if (time[3] == '?'){
            time[3] = '5';
        }
        if (time[4] == '?'){
            time[4] = '9';
        }
        return time;
    }
};
